libraries/c_sdk/standard/mqtt/test/unit/iot_tests_mqtt_subscription.c

```diff
@@ -217,6 +220,47 @@ static bool _waitForCount( IotMutex_t * pMutex,
 
 /*-----------------------------------------------------------*/
 
+/**
+ * @brief Wait for a reference count to reach a target value, using FreeRTOS semaphores,
+ * subject to a timeout.
+ */
+static bool _waitForCountSemaphore( StaticSemaphore_t * pSem,
```

Comment:
If the thought is to prefer use of native FreeRTOS synchronization primitives instead of `IotMutex`, can `_waitForCount` be removed from this file? It would be better to have a single approach used through the file instead of multiple functions present for a similar functionality..

Reply:
I have looked at that choice as well. This is for a mutex used by the library, not just by tests. If I need to update it, I will need to analyse and update that as well. I will keep that task as a different one to analyze and improve it if possible. It will be outside the scope of this PR.